packages/notifications/push/amplify_push_notifications_pinpoint/test/pinpoint_provider_test.dart

```diff
@@ -0,0 +1,213 @@
+    appId: 'APP_ID',
+    region: 'REGION',
+  );
+  final awsIamAmplifyAuthProvider = MockAWSIamAmplifyAuthProvider();
+  final mockAnalyticsClient = MockAnalyticsClient();
+  group('PinpointProvider', () {
+    test('init fails when retreiving an Auth provider was not successfull', () {
+      when(mockAmplifyAuthProviderRepository.getAuthProvider(any))
+          .thenReturn(null);
+      expect(
+        () async => pinpointProvider.init(
+          config: notificationsPinpointConfig,
+          authProviderRepo: mockAmplifyAuthProviderRepository,
+        ),
+        throwsA(
+          isA<ConfigurationError>().having(
+            (e) => e.message,
+            'config error',
+            contains('No AWSIamAmplifyAuthProvider available'),
+          ),
+        ),
+      );
+    });
+
+    test('identifyUser fails when the Pinpoint Provider is not initialized',
+        () async {
+      expect(
+        () async => pinpointProvider.identifyUser(
+          userId: 'userId',
+          userProfile: MockUserProfile(),
+        ),
+        throwsA(
+          isA<ConfigurationError>().having(
+            (e) => e.message,
+            'Not configured',
+            contains('Provider is not initialized'),
+          ),
+        ),
+      );
+    });
+
+    test(
+        'constructEventInfo should return journey data when there is journey details in the payload',
+        () async {
+      final res = pinpointProvider.constructEventInfo(
+        notification: PushNotificationMessage.fromJson(androidJourneyMessage),
+      );
+      final properties = res.properties;
+      final source = res.source;
+      expect(properties.attributes.containsKey('journey_id'), true);
+      expect(source, PinpointEventSource.journey.name);
+    });
+
+    test(
+        'constructEventInfo should return campaign data when there is campaign details in the payload',
+        () async {
+      final res = pinpointProvider.constructEventInfo(
+        notification: PushNotificationMessage.fromJson(androidCampaignMessage),
+      );
+      final properties = res.properties;
+      final source = res.source;
+      expect(properties.attributes.containsKey('campaign_id'), true);
+      expect(source, PinpointEventSource.campaign.name);
+    });
+  });
+
+  group('Happy path test', () {
+    TestWidgetsFlutterBinding.ensureInitialized();
+
+    test('init should run successfully', () async {
+      when(mockAmplifyAuthProviderRepository.getAuthProvider(any))
+          .thenReturn(awsIamAmplifyAuthProvider);
+      when(
+        mockAnalyticsClient.init(
+          pinpointAppId: anyNamed('pinpointAppId'),
+          region: anyNamed('region'),
+          authProvider: anyNamed('authProvider'),
+        ),
+      ).thenAnswer((realInvocation) async {});
+      await pinpointProvider.init(
+        config: notificationsPinpointConfig,
+        authProviderRepo: mockAmplifyAuthProviderRepository,
+        mockAnalyticsClient: mockAnalyticsClient,
+      );
```

Comment:
Use assertion to verify function calls can complete without exceptions throughout. 

```suggestion
      expect(
        pinpointProvider.init(
          config: notificationsPinpointConfig,
          authProviderRepo: mockAmplifyAuthProviderRepository,
          mockAnalyticsClient: mockAnalyticsClient,
        ),
        completes,
      );
```